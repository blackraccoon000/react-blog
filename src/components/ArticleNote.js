import React from "react"
import { DateTime } from "luxon"

const ArticleNote = (props) => {
  return (
    <div className="article-item">
      <figure className="article-item__photo">
        { props.link ?
          <img
            className="article-item__img"
            src={props.link}
            alt="dog"
          />
        :
          <div className="article-item__imgDiv">
            <p className="article-item__imgDiv__p">No Image</p>
          </div>
        }
      </figure>
      <div className="article-item__box">
        <h3 className="article-item__title">{props.title}</h3>
        <p className="article-item__text">{props.note}</p>
        <p className="article-item__date">{DateTime.fromMillis(props.createdAt).toFormat('yyyy LL dd')}</p>
      </div>
    </div>
  )
}

export default ArticleNote