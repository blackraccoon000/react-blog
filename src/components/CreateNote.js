import React from "react"
import { connect } from "react-redux"
import { DateTime } from "luxon";
import { createBlog, removeBlog, editBlog, firebaseAddBlog } from "../actions/blogActions"
import Image from "./Image"
import database from "../firebase/firebase"

class CreateNote extends React.Component {
  constructor(props){
    super(props)
    const id = props.match.params.id && props.match.params.id
    const title = props.findData.title && props.findData.title
    const note = props.findData.note && props.findData.note
    const link = props.findData.link && props.findData.link
    this.state = {
      id,
      title,
      note,
      error:"",
      link,
      images:[],
      imgFlag:false
    }
    console.log(this.state)
  }
  titleInput = (e) => {
    const title = e.target.value
    this.setState(()=>({ title }))
  }
  noteInput = (e) => {
    const note = e.target.value
    this.setState(()=>({ note }))
  }
  linkInput = (e) => {
    const link = e.target.src
    this.setState(()=>({ link }))
  }
  submit = (e) => {
    e.preventDefault()
    const { id, title, note, link } = this.state
    if( title==="" || note==="" ) {
      this.setState(()=>({ error: "titleとnoteに入力してください" }))
    } else {
      if(id) {
        this.props.editBlogD(id, { title, note, link })
        this.firebaseEdit(e)
      } else { // create
        this.firebaseAdd(e)
      }
      this.props.history.push("/")
    }
  }
  remove = (e) => {
    this.props.removeBlogD(this.state.id)
    this.firebaseRemove(e)
    this.props.history.push("/")
  }
  firebaseAdd = (e) => {
    e.preventDefault()
    const uid = process.env.FIREBASE_LOGIN_UID
    database.ref(`users/${uid}/`).push({
      title: this.state.title,
      note: this.state.note,
      link: this.state.link,
      createdAt: DateTime.now().ts
    }).then(res => {
      const { id=res.path.pieces_[2] , title, note, link } = this.state
      this.props.createBlogD({ id, title, note, link })
    })
    .catch(error => console.log(error))
  }
  firebaseEdit = (e) => {
    e.preventDefault()
    const uid = process.env.FIREBASE_LOGIN_UID
    const id = this.state.id
    const updates = {
      title:this.state.title,
      note: this.state.note,
      link: this.state.link,
      updatedAt: DateTime.now().ts
    }
    database.ref(`users/${uid}/${id}`).update(updates)
    .then(()=>{
      console.log("success")
    })
    .catch(error => console.log(error))
  }
  firebaseRemove = (e) => {
    e.preventDefault()
    const uid = process.env.FIREBASE_LOGIN_UID
    const id = this.state.id
    database.ref(`users/${uid}/${id}`).remove()
    .then(()=>{
      console.log("success")
    })
  }
  render() {
    return (
    <div className="note-container">
      {this.state.error && <p className="form__error">{this.state.error}</p>}
      <div className="edit-div">
        <form
          className="form-div"
        >
          { this.state.link
            ?
            <img src={this.state.link} alt="" className="sampleImg"/>
            :
            <div className="no-SampleImg">
              <p>No Image</p>
            </div>
          }
          <label
            htmlFor="title"
            className="label-title"
          >Title</label>
          <input type="text"
            className="edit-title"
            type="text"
            autoFocus
            placeholder="Titleを入力してください"
            onChange={this.titleInput}
            value={this.state.title}
            data-testid={"title"}
          />
          <label
            htmlFor="body"
            className="label-body"
          >Body</label>
          <textarea
            className="edit-body"
            placeholder="Note本文を入力してください"
            onChange={this.noteInput}
            value={this.state.note}
            data-testid={"note"}
          ></textarea>
        </form>
        <div className="btn-div">
          <button className="save-btn" onClick={this.submit}>save</button>
          {this.state.id &&
          <button className="remove-btn" onClick={this.remove} >remove</button>
          }
          <button
            className="img-btn"
            onClick={()=>{
              this.setState({imgFlag:!this.state.imgFlag})
            }}
          >image</button>
        </div>
        { this.state.imgFlag && <Image linkInput={this.linkInput}/>}
      </div>
    </div>
    )
  }
}

const mapStateToProps = (state,ownProps) =>{
  const findData = state.blogData.find(note => note.id === ownProps.match.params.id)
  console.log("get findData:",findData)
  return {
    findData: findData ? findData : {
      title:"",
      note:"",
      link:""
    },
    uid: state.auth.uid,
  }
}

const mapDispatchToProps = (dispatch) => ({
  createBlogD: (createData) => dispatch(createBlog(createData)),
  editBlogD: (id,updates) => dispatch(editBlog(id,updates)),
  removeBlogD: (id) => dispatch(removeBlog(id)),
})

export default connect(mapStateToProps, mapDispatchToProps)(CreateNote)
export { CreateNote }