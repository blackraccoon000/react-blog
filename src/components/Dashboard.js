import React from "react"

const Dashboard = (props) => {
  const history = props.history
  return (
    <div>
      <h1>This Component is Dashboard</h1>
      <button onClick={()=>{
        history.push("/")
        }}>Go Home</button>
    </div>
  )
}

export default Dashboard