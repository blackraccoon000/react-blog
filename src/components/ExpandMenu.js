import React from 'react';
import { Link } from 'react-router-dom';

class Translation extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      flag: false,
    };
  }
  onToggle = () => {
    this.setState({ flag: !this.state.flag });
  };
  render() {
    return (
      <div>
        <div id="cont">
          <div id="menu-fixed">
            <div class="notes">
              <p>Expand the menu!</p>
            </div>
            <a href="#cont">
              <i class="material-icons back">&#xE314;</i>
            </a>
            <a href="#menu-fixed">
              <div class="logo">
                <span></span>
                <p>TITLE</p>
              </div>
              <p class="pmenu">MENU</p>
            </a>
            <ul class="menu">
              <li>
                <i class="material-icons">&#xE88A;</i>
                <p>Home</p>
              </li>
              <li>
                <i class="material-icons">&#xE87A;</i>
                <p>Explore</p>
              </li>
              <li>
                <i class="material-icons">&#xE8CC;</i>
                <p>Shop</p>
              </li>
              <li>
                <i class="material-icons">&#xE8B8;</i>
                <p>Settings</p>
              </li>
              <li>
                <i class="material-icons">&#xE8B6;</i>
                <p>Search</p>
              </li>
            </ul>
            <i class="material-icons info">&#xE88E;</i>
          </div>
        </div>
        <div id="page">
          <div></div>
        </div>
      </div>
    );
  }
}

export default Translation;
