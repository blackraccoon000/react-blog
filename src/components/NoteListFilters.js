import React from "react"
import { connect } from "react-redux"
import { setText, setStartDate, setEndDate, setSqueezeNum, } from "../actions/filterActions"
import selectValueFilterFn from "../selectors/selectValueFilterFn"

const NoteListFilters = (props) => {
  return (
  <div>
    <input type="text" defaultValue={props.filters.text}
      onChange={props.inputTextFilter}
      data-testid="textFilter"/>
    <select
      value={props.filters.sortBy}
      onChange={props.selectValueFilter}
      data-testid="dataOrText"
    >
      <option value="date" data-testid="select-date">date</option>
      <option value="text" data-testid="select-text">text</option>
    </select>
    <select
      value={props.filters.sortOrder}
      onChange={props.selectValueFilter}
      data-testid="ascOrDes"
    >
      <option value="ascending">ascending</option>
      <option value="descending">descending</option>
    </select>
    <input type="number" defaultValue={props.filters.squeezeNum}
      onChange={props.inputSqueezeFilter}
      data-testid="squeezeFilter"
    />
  </div>
)}

const mapStateToProps = state => {
  return {
    filters: state.filters
  }
}

const mapDispatchToProps = dispatch => {
  return {
    inputTextFilter:(e) => dispatch(setText(e.target.value)),
    selectValueFilter:(e) => selectValueFilterFn(e,dispatch),
    inputSqueezeFilter:(e) => dispatch(setSqueezeNum(e.target.value))
  }
}

export default connect(mapStateToProps,mapDispatchToProps)(NoteListFilters)
export { NoteListFilters }