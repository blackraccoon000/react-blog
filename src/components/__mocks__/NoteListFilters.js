import React from "react"
const mock = jest.fn().mockImplementation(() => (<div>NoteListFilters Use ManualMock</div>))
export default mock