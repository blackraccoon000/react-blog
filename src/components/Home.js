import React from "react"
import { Link } from "react-router-dom"
import { connect } from "react-redux"
import ArticleNote from "./ArticleNote"
import getVisibleBlogData from "../selectors/getVisibleBlogData"

const Home = (props) => {
  return (
    <div className="content-container">
      {
        props.blogData.map(note => {
          return props.isAuthenticated ?
          <Link key={`link_${note.id}`} to={`/edit/${note.id}`}>
            <ArticleNote key={note.id} {...note} />
          </Link>
          :
          <ArticleNote key={note.id} {...note} />
        })
      }
    </div>
  )
}

const mapStateToProps = (state)=>{
  return {
    blogData: getVisibleBlogData(state.blogData,state.filters),
    isAuthenticated: !!state.auth.uid
  }
}

export default connect(mapStateToProps)(Home)
export { Home }