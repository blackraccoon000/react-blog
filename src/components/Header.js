import React from "react"
import NavMenu from "./NavMenu"
import { connect } from "react-redux"
import NoteListFilters from "./NoteListFilters"
class Header extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      flag:false
    }
  }
  onClickBtn = () => {
    this.setState({flag: !this.state.flag })
  }
  render() {
    return (
      <div>
        <header className="header">
          <div className="content-container">
            <h1 className="header__title">My Blog</h1>
            { this.state.flag && <NavMenu history={history}/> }
            { this.state.flag && <NoteListFilters/> }
            <button className="navBtn" data-testid="navBarFlagButton" onClick={this.onClickBtn}>
              {this.state.flag ? "Close" : "Open" }
            </button>
          </div>
        </header>
        <div className="header-alert">
          <div className="content-container">
            { this.props.isAuthenticated && "sign in now" }
          </div>
        </div>
      </div>
    )
  }
}

const mapStateToProps = state => ({
  isAuthenticated: !!state.auth.uid
})

export { Header }
export default connect( mapStateToProps )( Header )