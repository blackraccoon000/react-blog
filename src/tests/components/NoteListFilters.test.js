import React from "react"
import { render, fireEvent, screen, cleanup } from '@testing-library/react'
import { NoteListFilters } from "../../components/NoteListFilters"
import filters from "../fixtures/sampleFilters"

let wrapper, inputFilter, selectValueFilter, inputSqueezeFilter

beforeEach(()=>{
  inputFilter = jest.fn()
  selectValueFilter = jest.fn()
  inputSqueezeFilter = jest.fn()
  wrapper = render(<NoteListFilters
    inputFilter={inputFilter}
    selectValueFilter={selectValueFilter}
    inputSqueezeFilter={inputSqueezeFilter}
    filters={filters}
  />)
})

afterEach(()=>{
  cleanup
})

it("NoteListFilters Render", () => {
  expect(wrapper.container).toMatchSnapshot()
})

describe("Test ID find textFilter",()=>{
  it("textFilter to Be in the Document", () => {
    expect(screen.getByTestId("textFilter")).toBeInTheDocument()
  })
  it("textFilter to have attribute default value", () => {
    expect(screen.getByTestId("textFilter")).toHaveAttribute("value","filtersText")
  })
})

describe("Test ID find dataOrText",()=>{
  it("dataOrText to Be in the Document", () => {
    expect(screen.getByTestId("dataOrText")).toBeInTheDocument()
  })
  it("dataOrText to have attribute default value", () => {
    expect(screen.getByTestId("dataOrText")).toHaveDisplayValue("date")
  })
  /**
   * select の内容を変更するテストを書きたかったが、失敗した。
   */
  // it("dataOrText select value change Testing", () => {
    // console.log(wrapper.getByDisplayValue("date"))
    // fireEvent.change(wrapper.getByDisplayValue("date"),{ target: { value: 'text' } })
    // expect(wrapper.getByTestId("dataOrText")).toHaveDisplayValue("text")
  // })
})

describe("Test ID find ascOrDes",()=>{
  it("ascOrDes to Be in the Document", () => {
    expect(screen.getByTestId("ascOrDes")).toBeInTheDocument()
  })
  it("ascOrDes to have attribute default value", () => {
    expect(screen.getByTestId("ascOrDes")).toHaveDisplayValue("ascending")
  })
})
