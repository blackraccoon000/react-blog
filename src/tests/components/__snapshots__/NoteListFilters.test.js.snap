// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteListFilters Render 1`] = `
<div>
  <div>
    <input
      data-testid="textFilter"
      type="text"
      value="filtersText"
    />
    <select
      data-testid="dataOrText"
    >
      <option
        data-testid="select-date"
        value="date"
      >
        date
      </option>
      <option
        data-testid="select-text"
        value="text"
      >
        text
      </option>
    </select>
    <select
      data-testid="ascOrDes"
    >
      <option
        value="ascending"
      >
        ascending
      </option>
      <option
        value="descending"
      >
        descending
      </option>
    </select>
    <input
      data-testid="squeezeFilter"
      type="number"
      value=""
    />
  </div>
</div>
`;
